app/models/solidus_paypal_braintree/gateway.rb

```diff
@@ -261,13 +261,31 @@ def transaction_options(source, options, submit_for_settlement = false)
         params[:payment_method_nonce] = source.nonce
       end
 
+      if source.paypal?
+        params[:shipping] = braintree_shipping_address(options)
+      end
+
       if source.customer.present?
         params[:customer_id] = source.customer.braintree_customer_id
       end
 
       params
     end
 
+    def braintree_shipping_address(options)
+      address = options[:shipping_address]
+      first, last = address[:name].split(" ", 2)
+      {
+        first_name: first,
+        last_name: last,
+        street_address: address[:address1].to_s + address[:address2].to_s,
```

Comment:
You should add a space in between. Otherwise

```ruby
{
  address1: "Street 1",
  address2: "c/o Stembolt"
}
```

would become "Street1c/o Stembolt"